packages/react-core/src/components/MenuToggle/MenuToggleCheckbox.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Check/check';
+import { css } from '@patternfly/react-styles';
+import { PickOptional } from '../../helpers/typeUtils';
+import { getOUIAProps, OUIAProps, getDefaultOUIAId } from '../../helpers';
+
+export interface MenuToggleCheckboxProps
+  extends Omit<React.HTMLProps<HTMLInputElement>, 'type' | 'onChange' | 'disabled' | 'checked'>,
+    OUIAProps {
+  /** Additional classes added to the MenuToggleCheckbox */
+  className?: string;
+  /** Flag to show if the checkbox selection is valid or invalid */
+  isValid?: boolean;
+  /** Flag to show if the checkbox is disabled */
+  isDisabled?: boolean;
+  /** Flag to show if the checkbox is checked */
+  isChecked?: boolean | null;
+  /** Alternate Flag to show if the checkbox is checked */
+  checked?: boolean | null;
```

Comment:
Because there's been issues with other components that have both an `isChecked` and `checked` prop, I'm wondering if it would be better to have `isChecked` be described as the prop to use when you want it to be controlled, and `checked` to be the prop to use when you want it to be uncontrolled.

If we go this route, then further down we would want to have the `checked` attribute on the input element = `isChecked`, and the `defaultChecked` attribute = `checked`. [React#7557](https://github.com/patternfly/patternfly-react/pull/7557/files) has a good example of this.

@nicolethoen wdyt?

Reply:
oof - this always makes my head spin.

I honestly think it's clearer to use `isChecked` when you want a controlled check state and `defaultChecked` when you want an uncontrolled state.

I guess it's true that it is not completely consistent with other components, but it feels clearer (if it's correct - like i said, this makes my head spin).

Reply:
> I honestly think it's clearer to use isChecked when you want a controlled check state and defaultChecked when you want an uncontrolled state.

I agree

>I guess it's true that it is not completely consistent with other components, but it feels clearer (if it's correct - like i said, this makes my head spin).

I think other components would eventually need to be updated, especially ones where `checked` and `isChecked` are being used for the same thing. I think this is a good opportunity to avoid that issue down the line.

Reply:
@tlabaj do you have any preferences? Can we use `isChecked` and `defaultChecked` for the controlled/uncontrolled states rather than `isChecked` and `checked`?

Reply:
`checked` and `isChecked` are technically the same.  They both should set the `checked` attribute on the input.  I think when we have the next major release, we can update the code to remove the `checked` prop. If the consumer does decide to use `checked` rather than `isChecked` it will still work since we are spreading the props on the input.  I don't think any of our components use the `checked` prop to set `defaultChecked` attribute.